macros/finetune_macro.C

```diff
@@ -0,0 +1,91 @@
+std::map<int, std::string> device_int2string = {
+    {192, "kc705-192"},
+    {193, "kc705-193"},
+    {194, "kc705-194"},
+    {195, "kc705-195"},
+    {196, "kc705-196"},
+    {197, "kc705-197"},
+    {198, "kc705-198"},
+    {207, "kc705-207"}};
+
+std::map<std::string, int> device_string2int = {
+    {"kc705-192", 192},
+    {"kc705-193", 193},
+    {"kc705-194", 194},
+    {"kc705-195", 195},
+    {"kc705-196", 196},
+    {"kc705-197", 197},
+    {"kc705-198", 198},
+    {"kc705-207", 207}};
+
+TF1 *fine_analysis_fit_function = new TF1("fine_analysis_fit_function", "[0]*(1./(exp((x-[1])/[2])+1))*(1./(exp(([3]-x)/[4])+1))", 10., 200.);
+
+void finetune_macro(std::string filename)
+{
+  // Output
+  std::map<std::string, TH1F *> _TH1F;
+
+  //  Open file with TDC fine distributions
+  TFile *input_file = new TFile(filename.c_str());
+  std::map<std::string, TH2F *> input_TDC_dist;
+  for (auto [current_device_index, current_device] : device_int2string)
+  {
+    //  Take 2D TDC distribution for fit
+    TH2F *current_calib_histo = (TH2F *)(input_file->Get(Form("hFine_%i", current_device_index)));
+    if (!current_calib_histo)
+      continue;
+    input_TDC_dist[current_device] = (TH2F *)current_calib_histo->Clone();
+  }
+  //  Loop over available TDC distributions
```

Comment:
Useless complication, loop over all possible devices. If input file does not exist you do nothing. No need of storing something and duplicate the loops